packages/hardhat-ethers/test/no-account.test.ts

```diff
@@ -0,0 +1,129 @@
+import { Artifact, HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { SignerWithAddress } from "../src/signers";
+
+import { useEnvironment } from "./helpers";
+
+describe("hardhat-ethers plugin", function () {
+  describe("hardhat network with no accounts", function () {
+    useEnvironment("hardhat-project-no-accounts", "hardhat");
+
+    describe("fixture setup", function () {
+      it("should not have accounts", async function () {
+        const signers = await this.env.ethers.getSigners();
+        assert.isEmpty(signers);
+      });
+    });
+
+    describe("getContractAt", function () {
+      let signerAddress: string;
+
+      beforeEach(async function () {
+        await this.env.network.provider.request({
+          method: "evm_mine",
+          params: [],
+        });
+
+        const { miner } = await this.env.ethers.provider.getBlock("latest");
+        signerAddress = miner;
+
+        await this.env.run(TASK_COMPILE, { quiet: true });
+      });
+
+      describe("with the name and address", function () {
+        it("Should return an instance of a contract with a read-only provider", async function () {
+          const receipt = await deployGreeter(this.env, signerAddress);
+
+          assert.isTrue(this.env.ethers.provider._isProvider);
+          const contract = await this.env.ethers.getContractAt(
+            "Greeter",
+            receipt.contractAddress
+          );
+
+          assert.isDefined(contract.provider);
+          assert.isNotNull(contract.provider);
+          assert.strictEqual(contract.provider, this.env.ethers.provider);
+          assert.containsAllKeys(contract.functions, [
+            "setGreeting(string)",
+            "greet()",
+          ]);
+
+          const greeting = await contract.functions.greet();
+
+          assert.equal(greeting, "Hi");
+        });
+      });
+
+      describe("with the abi and address", function () {
+        it("Should return an instance of a contract with a read-only provider", async function () {
+          const receipt = await deployGreeter(this.env, signerAddress);
+
+          const signers = await this.env.ethers.getSigners();
+          assert.isEmpty(signers);
+
+          const greeterArtifact = await this.env.artifacts.readArtifact(
+            "Greeter"
+          );
+
+          const contract = await this.env.ethers.getContractAt(
+            greeterArtifact.abi,
+            receipt.contractAddress
+          );
+
+          assert.isDefined(contract.provider);
+          assert.isNotNull(contract.provider);
+          assert.strictEqual(contract.provider, this.env.ethers.provider);
+          assert.containsAllKeys(contract.functions, [
+            "setGreeting(string)",
+            "greet()",
+          ]);
+
+          const greeting = await contract.functions.greet();
+
+          assert.equal(greeting, "Hi");
+        });
+      });
+    });
+
+    describe("getSigner", function () {
```

Comment:
This test is out of place here.

Reply:
I mean in this file, named 'no-account'.

Reply:
I actually thought it could go here since it ensures that this account doesn't exist in the network.

Reply:
I see your reasoning now, but that's not an actual precondition of the function, nor it's tested.

Reply:
Hm, that's true. I can add an assert for the non existence of the account.

It's not a precondition, but it should be the most commonly used case. I.e. when impersonating an account, that account won't exist. I'm not sure if adding a test for an existing account is useful though, since as you say, it's not a precondition really.